src/ir/ty.rs

```diff
@@ -468,7 +468,7 @@ impl Type {
                                 return Err(ParseError::Recurse);
                             }
 
-                            error!("invalid type {:?}", ty);
+                            warn!("invalid type {:?}", ty);
```

Comment:
This *can* be an error, but it is not if `ty.spelling().is_empty()`, so I think we want to wrap that in an `if`.